src/config.c

```diff
@@ -64,26 +64,54 @@ static int ticket_realloc(void)
 
 static void hostname_to_ip(char * hostname)
 {
-	struct hostent *he;
-	struct in_addr **addr_list;
+	struct addrinfo hints;
+	struct addrinfo *result, *rp;
+	int res;
+	int addr_found = 0;
+	const char *ntop_res;
 
-	if ((he = gethostbyname(hostname)) == NULL) {
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = AF_UNSPEC;
+	hints.ai_socktype = SOCK_STREAM;
+
+	res = getaddrinfo(hostname, NULL, &hints, &result);
+
+	if (res != 0) {
 		log_error("can't find IP for the host \"%s\"", hostname);
 		return;
 	}
 
-	addr_list = (struct in_addr **) he->h_addr_list;
+	/* Return the first found AF_INET or AF_INET6 address */
+	for (rp = result; rp != NULL && !addr_found; rp = rp->ai_next) {
```

Comment:
To me it's always easier to read "if (rp && ...)" rather than "if (rp != NULL && ...)"
A matter of opinion, of course, but whenever I see "if (p)" I automatically know what is meant.

Reply:
Changed. Honestly (as you could seen) I'm more fan of explicit == NULL (so I can clearly see if variable is boolean or pointer), but it's more important to keep code consistent and booth is using shorter version so I've changed this line and also `if (ntop_res != NULL) {` line.